packages/api/src/cli/codegen/languages/typescript.ts

```diff
@@ -566,10 +567,10 @@ sdk.server('https://eu.api.example.com/v14');`),
 
     let hasOptionalBody = false;
     let hasOptionalMetadata = false;
-    const parameters: {
-      body?: OptionalKind<ParameterDeclarationStructure>;
-      metadata?: OptionalKind<ParameterDeclarationStructure>;
-    } = {};
+    const parameters = {} as {
+      body: OptionalKind<ParameterDeclarationStructure>;
+      metadata: OptionalKind<ParameterDeclarationStructure>;
+    };
```

Comment:
This slight reshuffling ended up fixing all of those `ts-morph` typing issues. When we use `parameters.body` and `parameters.metadata` they will be present so marking them as optional here is wrong. I believe I had originally done it this way to satisfy setting `parameters` to an empty object.